File: src/ast/CtorDtorNameType.java
package ast;

import java.io.StringWriter;

public class CtorDtorNameType extends ParentNode {
    private boolean _isDestructor;

    public CtorDtorNameType(BaseNode name, boolean isDestructor) {
        super(NodeType.CtorDtorNameType, name);
        _isDestructor = isDestructor;
    }

    @Override
    public void PrintLeft(StringWriter writer) {
        if (_isDestructor) {
            writer.write("~");
        }

        writer.write(child.GetName());
    }
}

File: src/de/fearlesstobi/demangler/ast/CtorDtorNameType.kt
package de.fearlesstobi.demangler.ast

import java.io.StringWriter

class CtorDtorNameType(name: BaseNode?, private val isDestructor: Boolean) : ParentNode(NodeType.CtorDtorNameType, name) {
    public override fun printLeft(writer: StringWriter) {
        if (isDestructor) {
            writer.write("~")
        }
        writer.write(child!!.getName())
    }

}